### PR TITLE
ci: use composite action to minimize code duplication

### PR DESCRIPTION
**Pull request by zephvr**
#### Describe your PR, what does it fix/add?
The idea behind this code refactoring is to factorize the image setup process used in CI. The goal is also to make it more readable and easily maintainable

GitHub composite action seemed like a good choice for this purpose, as it implement in between steps. I also choose local action compare to remote for better control.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
I also changed the build directory for meson from `obj-x86_64-pc-linux-gnu` to `build`. I did not find any reason for the old name, while the new is explicitly given in the contributing guide.

I also updated the `actions/checkout` to v4

#### Is it ready for merging, or does it need work?
Is ready for merging

### CODE REVIEW

File: .github/actions/setup_base/action.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          meson \
+          ninja \
+          pango \
+          pixman \
+          pkgconf \
+          scdoc \
+          seatd \
+          systemd \
+          tomlplusplus \
+          vulkan-headers \
```
Comment (vaxerski) [inline .github/actions/setup_base/action.yml:48]: if we're refactoring, vulkan deps can be dropped

File: .github/actions/setup_base/action.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          xorg-xkbcomp \
+          xorg-xwayland
+
+    - name: Set up user
+      shell: bash
+      run: |
+        useradd -m githubuser
+        echo -e "root ALL=(ALL:ALL) ALL\ngithubuser ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers
+
+    - name: Install libdisplay-info from the AUR
```
Comment (vaxerski) [inline .github/actions/setup_base/action.yml:73]: same as above, it's now in official pkgs

**Comment by zephvr** [comment]
I just did what you requested. Should be good now

**Review by vaxerski — APPROVED** [review]
lgtm, I dont speak CI well but it should work

**Comment by vaxerski** [comment]
thanks!

**Pull request merged**